### DISCUSSION TITLE
estimate_parasitics -global_routing on a post-global_route design loaded via read_db

### DISCUSSION THREAD

**Discussion by nayiri-k** [The-OpenROAD-Project/OpenROAD#2880] [General]
Is it possible to run `estimate_parasitics -global_routing` on a post-global_route design that's been loaded via the `read_db` command? As in, `global_route` has been run on the design before it was saved with `write_db`, but `global_route` has not been run in this current OpenROAD session. I keep getting this error:

```
>>> read_db post_global_route
>>> estimate_parasitics -global_routing
[ERROR RSZ-0005] Run global_route before estimating parasitics for global routing.
```

**Comment by maliberty** [comment]
@eder-matheus I thought this was possible.  Please comment

**Reply by eder-matheus** [reply]
Yes, it is possible. As long as you have the guides in the database, global route should be able to load them to its internal structures when calling `estimate_parasitics`.
@nayiri-k  A testcase with the failure would be good, if possible.

**Reply by eder-matheus** [reply]
@nayiri-k You can also try to read the guides file before running `estimate_parasitics`, if you have written it in the global route run.

**Reply by nayiri-k** [reply]
The `read_guides` command fixed it, I wasn't aware of it before. Thanks!

**Comment by maliberty** [comment]
Are you using a very old version of OR for this?  The guides should be saved in the db.

**Reply by nayiri-k** [reply]
Yes you're right my version was a few months old, new version doesn't need the `read_guides` command.

**Comment by maliberty** [comment]
OR improves a lot over time so it is a good practice to test with a current version when you find a problem.
